src/components/Layout/BurgerMenu.tsx

```diff
@@ -1,22 +1,42 @@
-import React, { useContext } from 'react'
-import { BurgerMenuShowContext } from '../Context/BurgerMenuShowContext'
+import React, { useContext, useMemo} from 'react'
 import { BurgerMenuStatusContext } from '../Context/BurgerMenuStatusContext'
 import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
 import './BurgerMenu.css'
 
 function BurgerMenu() {
-  const statusContext = useContext(BurgerMenuStatusContext)
-  const showContext = useContext(BurgerMenuShowContext)
+  const menuContext = useContext(BurgerMenuStatusContext)
+  const [show, open, searching, filtering] = useMemo(() => {
+    return [
+      menuContext?.status.show,
+      menuContext?.status.open,
+      menuContext?.status.searching,
+      menuContext?.status.filtering]
+  }, [menuContext?.status])
 
-  const toggleMenuHandler = () => {
-    statusContext?.setStatus(prev => !prev)
+  const handleClick = () => {
+    if (!searching) {
+      menuContext?.setStatus(({ ...menuContext.status,
```

Comment:
As setStatus is a "setState" from useState, you can use here the previous state from setState instead of using the state. It may cause an issue if status is updated and this function didn't catch up with this update (it's not wrapped in a useCallback).
E.g.: `menuContext.setStatus(prevState => ({ ...prevState, open, filtering })`